src/mystery_gift.c

```diff
@@ -0,0 +1,293 @@
+#include "global.h"
+#include "mystery_gift.h"
+#include "save.h"
+#include "save_arrays.h"
+#include "constants/save_arrays.h"
+
+BOOL MysteryGiftTagIsValid(u32 tag);
```

Comment:
MysteryGift_TagIsValid

Reply:
This name implies the argument is a MysteryGift*, which it isn't.

Reply:
no? the name implies that the function is to do with mystery gifts, I thought that was how the naming is done, not based on arguments

Reply:
I feel like further discussion is needed on what the prefixes are based on exactly, so it's down and agreed

Reply:
Pika is right about what we based this nomenclature on.